auth/rest.go

```diff
@@ -207,3 +242,121 @@ func setCORSHeaders(headers http.Header) {
 	headers.Set("Access-Control-Allow-Origin", "*")
 	headers.Set("Access-Control-Allow-Credentials", "true")
 }
+
+func rewriteAccessTokens(resp *httptest.ResponseRecorder, req *http.Request, a *RestAuthDecorator) error {
+	err := rewriteBodyAccessTokens(resp, req, a)
+	if err != nil {
+		return err
+	}
+
+	err = rewriteHeaderAccessTokens(resp, req, a)
+	if err != nil {
+		return err
+	}
+
+	return rewriteCookieAccessTokens(resp, req, a)
+}
+
+func rewriteBodyAccessTokens(resp *httptest.ResponseRecorder, req *http.Request, a *RestAuthDecorator) error {
+	// rewrite body tokens
+	bodyTokenKey := resp.Header().Get("X-Gateway-BodyToken")
+	if bodyTokenKey != "" {
+
+		var response map[string]interface{}
+		jsonBlob, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		err = json.Unmarshal(jsonBlob, &response)
+
+		if err != nil {
+			return err
+		}
+
+		bodyToken, ok := response[bodyTokenKey]
+
+		if !ok {
+			return err
+		}
+
+		jwtResponse := JWTResponse{}
+		jwtResponse.JWT = bodyToken.(string)
+
+		token, _, err := a.tokenStore.AddToken(&jwtResponse)
+		if err != nil {
+			return err
+		}
+
+		response[bodyTokenKey] = token
+		jsonResponse, err := json.Marshal(&response)
+		if err != nil {
+			return err
+		}
+		resp.Write(jsonResponse)
```

Comment:
Are you sure that `resp.Write` does _overwrite_ the existing body, and does not _append_ to it?

Reply:
Well it did overwrite it in my tests locally.